integreat_compass/cms/forms/custom_model_form.py

```diff
@@ -0,0 +1,188 @@
+        if self.instance.id:
+            attributes.append("instance")
+
+        self.logger.debug(
+            "%s initialized"
+            + (" with " + ": %r, ".join(attributes) + ": %r" if attributes else ""),
+            type(self).__name__,
+            *[getattr(self, attribute) for attribute in attributes],
+        )
+
+        for field_name, field in self.fields.items():
+            # If form is disabled, disable all form fields
+            if disabled:
+                field.disabled = True
+
+            # Set placeholder for every text input fields
+            if isinstance(
+                field.widget,
+                (
+                    forms.TextInput,
+                    forms.Textarea,
+                    forms.EmailInput,
+                    forms.URLInput,
+                    forms.PasswordInput,
+                    forms.NumberInput,
+                ),
+            ):
+                try:
+                    # Use verbose_name of model field instead of field label because label is capitalized
+                    # pylint: disable=no-member
+                    model_field = self._meta.model._meta.get_field(
+                        field_name
+                    ).verbose_name
+                except FieldDoesNotExist:
+                    # In case field is not a model field, just use the label and make it lowercase
+                    model_field = lowfirst(field.label)
+
+                field.widget.attrs.update(
+                    {"placeholder": capfirst(_("Enter {} here").format(model_field))}
+                )
+
+    def clean(self):
+        """
+        This method extends the default ``clean()``-method of the base :class:`~django.forms.ModelForm` to provide debug
+        logging
+
+        :return: The cleaned data (see :ref:`overriding-modelform-clean-method`)
+        :rtype: dict
+        """
+        # Validate ModelForm
+        cleaned_data = super().clean()
+        self.logger.debug(
+            "%s validated with cleaned data %r", type(self).__name__, cleaned_data
+        )
+        return cleaned_data
+
+    def save(self, commit=True):
+        """
+        This method extends the default ``save()``-method of the base :class:`~django.forms.ModelForm` to provide debug
+        logging
+
+        :param commit: Whether or not the changes should be written to the database
+        :type commit: bool
+
+        :return: The saved object returned by :ref:`django:topics-modelform-save`
+        :rtype: object
+        """
+        self.logger.debug(
+            "%s saved with changed data %r", type(self).__name__, self.changed_data
+        )
+        # Update submitted data with cleaned data if changes are saved
+        self.data = self.cleaned_data
+
+        # Save ModelForm, add creator if required
+        instance = super().save(commit=False)
+        if hasattr(self.instance, "user"):
+            instance.creator = self.instance.user
```

Comment:
What exactly is the reason for the different names of the attributes?
Why set `instance.user = request.user` and then later `instance.creator = instance.user`? Why not directly `instance.creator = request.user` via `additional_instance_attributes`?

Reply:
> `additional_instance_attributes`

that was my initial idea as well, but I could not get this to work correctly.

I think it's just that there is no instance initially.

Reply:
In theory, the base class should automatically create an instance object if it doesn't really exist. The only difference of the new instance is that it does not yet have an id, but setting attributes should work even before it's saved to the database... or is the problem that the creator is a related field that cannot be assigned before saving? 🤔
And don't we use the same pattern for integreat-cms as well?

Reply:
I'll have another go at it, maybe the issue lay elsewhere. But neither `additional_instance_attributes` nor setting the field during `clean` worked, the only place I got it working was after saving without committing, so I think you might be onto something re: related field.

If you do  not think it makes sense to have this in the `CustomModelForm` though, we could of course just do it in the `OfferFormView`.

Reply:
> additional_instance_attributes

No idea what caused my previos issue, now it works as intended 😅